firestore/runtime/src/main/java/io/quarkiverse/googlecloudservices/firestore/runtime/FirestoreProducer.java

```diff
@@ -68,4 +77,16 @@ private Duration convertDuration(java.time.Duration duration) {
         return Duration.ofMillis(duration.toMillis());
     }
 
+    /**
+     * Determine if we need to use emulator credentials. Emulator credentials are used in case the host-override is set
+     * and we don't have accessTokens enabled.
+     *
+     * @return whether to use the emulator credentials
+     */
+    private boolean useEmulatorCredentials() {
+        return !this.gcpBootstrapConfiguration.accessTokenEnabled()
+                && this.firestoreConfiguration.hostOverride.isPresent()
+                && this.firestoreConfiguration.hostOverride.get().contains("localhost");
```

Comment:
Why also checking on localhost?
If an emulator is used for ex on a remote Docker engine, or if the user uses multiple host on its machine it can be on something else than localhost.

Reply:
Ref: com.google.cloud.firestore.spi.v1.GrpcFirestoreRpc:90

Reply:
OK, if it is the same inside the Google Cloud library so let's keep it this way, we can always relax it later if needed.